plotter/functions.go

```diff
@@ -39,7 +41,20 @@ func (f *Function) Plot(c draw.Canvas, p *plot.Plot) {
 		line[i].X = trX(x)
 		line[i].Y = trY(f.F(x))
 	}
-	c.StrokeLines(f.LineStyle, c.ClipLinesXY(line)...)
+
+	// For every continuous block of non-NaN Y values, stroke lines
```

Comment:
Is it possible to roll this into the previous loop? I'm thinking about something like this:
```
var line []draw.Point
for i := 0; i < f.Samples; i++ {
	x := p.X.Min + float64(i)*d
	y := f.F(x)
	if !math.IsNan(y) {
		line = append(line, draw.Point{X: trX(x), Y: trY(Y) })
	}
	// If we are at the end of a contiguous segment, stroke.
	if (math.IsNan(y) || i == f.Samples-1) {
		if len(line) > 1 {
			c.StrokeLines(f.LineStyle, c.ClipLinesXY(line)...)
		}
		line = line[:0]
	}
}
```